azure/packages/azure-client/src/test/types/validateAzureClientPrevious.generated.ts

```diff
@@ -349,54 +349,6 @@ declare function use_old_TypeAliasDeclaration_IAzureAudience(
 use_old_TypeAliasDeclaration_IAzureAudience(
     get_current_TypeAliasDeclaration_IAzureAudience());
 
-/*
-* Validate forward compat by using old type in place of current type
-* If breaking change required, add in package.json under typeValidation.broken:
-* "InterfaceDeclaration_ITelemetryBaseEvent": {"forwardCompat": false}
-*/
-declare function get_old_InterfaceDeclaration_ITelemetryBaseEvent():
```

Comment:
this doesn't make any sense to me. it seems like this should be a breaking change if these are disappearing.

Reply:
This does not make any sense to me either, the azure-client change I made was very small, change common-definitions to core-interfaces; https://github.com/microsoft/FluidFramework/pull/16211/files#diff-a407348d6eec77ff2ce19dd0793ceda56448c585efc52dadf0ed1715fed8adfbR5

Reply:
according to the API reports you made changes to the public API of 6 packages. I'd expect the same changes to be reflected in the type tests for those packages, and possibly some packages that use them.

Reply:
This specific case looks like something more is going on here. Currently we arn't supper clear on what type tests are actually supposed to do (particularly related to types which are reexported via other APIs), but it seems like for some reason the type tests in this package used to include testing that the current and previous versions of ITelemetryBaseEvent were compatible. Now it's no longer testing that. Since ITelemetryBaseEvent is not defined in this package, and you updated how its defined in the package it is imported from, something about the detail of how its defined in the package its imported from apparently impacts how the type test generation decided to test it or not. This is odd, but prabably fine.

Reply:
If you want to look deeper, "getNodeTypeData" in typeData.ts is where the logic for enumerating types including in a public API lives, that is what prabably handled these two cases differently.

Reply:
it might be wise to split this pr apart. get in the obvious stuff, and then work on these, rather than block it all.

Reply:
i don't know if i agree with the reasoning here. type tests test the exported surface area of a package. it seems like that shouldn't change. are these types explicitly exported from the package? Do they continue to be exported?

Reply:
Thanks @CraigMacomber! We'll add a follow up item to investigate why the type tests appear inconsistent with respect to re-exported APIs.

Reply:
@anthony-murphy the types tested here are not defined in this package (azure client does not define ITelemetryBaseEvent). I suspect they are used in the public API of some typed exported by this package (which themselves have type tests). Its not clear to me if type tests are supposed to transitively test all types referenced by the public API individually, or just test the types defined in this package: apparently the implementation of the type test of this policy choice seems to depend on stuff in the package they are imported from (ex: maybe it handles reexported types different from types defined where what are imported from?). Given that I don't know which behavior is correct, or if we care, I'm not sure if there is a bug here, and if so which behaviour is the bug. Regardless I don't think this PR is at has a problem for removing type tests of types which already have type tests already in another package: that seems safe.

Reply:
@CraigMacomber  they are explicitly exports, so are part of this packages api, it's not just a transitive reference:
https://github.com/microsoft/FluidFramework/blob/b57cd6ad3c92405175b0c3af872bda05338555c7/azure/packages/azure-client/src/interfaces.ts#L13

Reply:
before this change. consumer could do `import { ITelemetryBaseEvent} from "@fluidframework/azure-client"` does that still work?

Reply:
there are no code changes to this package, and its API report (which I trust far more than our type tests) also indicates no changes to its API, so it should be unchanged.

Reply:
that is a great point. and definitely increases confidence that there isn't anything bad actually happening here.